awx/main/tasks.py

```diff
@@ -934,6 +941,11 @@ def build_env(self, instance, private_data_dir, isolated, private_data_files=Non
         if self.should_use_proot(instance):
             env['PROOT_TMP_DIR'] = settings.AWX_PROOT_BASE_PATH
         env['AWX_PRIVATE_DATA_DIR'] = private_data_dir
+
+        if 'ANSIBLE_COLLECTIONS_PATHS' in env:
+            env['ANSIBLE_COLLECTIONS_PATHS'] += os.pathsep + settings.AWX_ANSIBLE_COLLECTIONS_PATHS
+        else:
+            env['ANSIBLE_COLLECTIONS_PATHS'] = settings.AWX_ANSIBLE_COLLECTIONS_PATHS
```

Comment:
Let me make the obligatory point that every time we do this, we squash the user's configuration

https://github.com/ansible/awx/issues/2973

By whitelisting the directory `/var/lib/awx/collections` by default, we prevent users from setting this up manually using only the Ansible core documentation / instructions. We change it so that they are required to file AWX-specific instructions.

It is nice that any existing env var that sets this is respected, but that seems much less likely (I don't easily see how it could happen) than trying to use some ansible.cfg, which cannot be respected with this method.

Reply:
It's not super obvious, but we _do_ respect users evn variables _if_ they are set via our configure tower in tower setting `AWX_TASK_ENV`.

So for the case at hand. If the user sets `ANSIBLE_COLLECTION_PATHS` via `AWX_TASK_ENV` then `env` will be populated when we perform the if else in this changset. If set, the users paths will be preserved.

Reply:
AWX_TASK_ENV is always a last resort hammer. We shouldn't add more things that force use of it.

Reply:
I would think of this as a hammer scenario. We are giving them a dir to put collections in `/var/lib/awx/collections`. If they want other dirs mapped in, they can use the hammer. We don't necessarily want users doing this. We want users using requirements.yml file in a project.

Reply:
This was meant to be more of an informational "known limitation" kind of comment. This setting is just carrying forward the current standard practice.